src/cares_wrap.cc

```diff
@@ -1565,6 +1565,8 @@ Maybe<int> SoaTraits::Parse(QuerySoaWrap* wrap,
 
   if (status != ARES_SUCCESS) return Just<int>(status);
 
+  auto cleanup = OnScopeLeave([&]() { ares_free_data(soa_out); });
```

Comment:
As a follow-up TODO here (doesn't need to be done in this PR) .. it would likely be good to come up with a smart-pointer for these.